### PR TITLE
Refactor regex code, add liquid whitespace control compatibility

### PR DESCRIPTION
**Pull request by lucascosti**
This PR originally started out as me wanting to add the ability for the extension to open reusabales/feature versioning tags that contained whitespace control.

As the extension has evolved over time, there were many places where similar regexes are used, which made changes like this difficult.

What originally started as me consolidating regexes eventually turned into this PR, which refactors a lot of that code. 😅 

This should make future updates (including possibly adding new things to open) hopefully much easier. 🎉

I've made a lot of code comments to explain what each thing does for future maintainers (and future me). I'd be happy to walk through the changes for anyone interested!

I've done a lot of testing, but please test this extensively 🙂

### CODE REVIEW

File: copier.js [hunk 1]
```diff
@@ -6,68 +6,20 @@ function copyMain() {
...
-            by: "character",
-            value: moveLeftBy
-        })  
-        // Move the cursor to the end of the reusable, selecting the text moved over
-        vscode.commands.executeCommand("cursorMove", {
-            to: "right",
-            by: "character",
-            value: moveRightBy,
-            select: true
-        })
```
Comment (lucascosti) [inline copier.js:69]: After figuring out ranges for the reusables in the `shared` file, thiscode  turned out to be unnecessary to select the thing after the copy. The new code below (with the help of Copilot) is much more simple and reliable 🚀

File: shared.js [hunk 2]
```diff
@@ -9,20 +9,104 @@ function testNoTab(editor) {
...
-    if (objSelectTextAroundCursor) {
-        reusableString = editor.document.getText(objSelectTextAroundCursor); 
+// This regex matches liquid tags. e.g:
+// - {%<something>%}
+const liquidTagRegex = /{%([^%])*%}/;
+
+// This regex matches frontmatter feature flags definitions. e.g.:
+// - feature: '<something>'
+// - feature: <something>
+const frontmatterFeatureFlagRegex = /^ *feature: '*([^ ']*)'* *$/;
```
Comment (lucascosti) [inline shared.js:19]: These are now the only regex definitions for liquid tags and frontmatter feature flags.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,14 @@
 # Change log
 
+## 1.6.0
+2 December 2022
+
+Major refactor of the code by @lucascosti to faciliate future changes to this extension.
```
Comment (hubwriter) [inline CHANGELOG.md:6]:
```suggestion
Major refactor of the code by @lucascosti to facilitate future changes to this extension. 
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,14 @@
 # Change log
 
+## 1.6.0
+2 December 2022
+
+Major refactor of the code by @lucascosti to faciliate future changes to this extension. 
+
+This version:
+- Adds the ability for the extension to open reusabales/feature versioning tags that contain whitespace control.
```
Comment (hubwriter) [inline CHANGELOG.md:9]:
```suggestion
- Adds the ability for the extension to open reusables/feature versioning tags that contain whitespace control.
```

**Review by Yedfree — APPROVED** [review]

**Pull request merged**